pkg/minikube/node/config.go

```diff
@@ -156,6 +152,32 @@ func setupKubeconfig(h *host.Host, c *config.ClusterConfig, n *config.Node, clus
 	return kcs, nil
 }
 
+// apiServerURL returns a URL to end user can reach to the api server
+func apiServerURL(h host.Host, cc config.ClusterConfig, n config.Node) (string, error) {
+	if driver.IsKIC(h.DriverName) {
+		p, err := oci.HostPortBinding(h.DriverName, h.Name, n.Port)
+		if err != nil {
+			return "", errors.Wrap(err, "host port binding")
+		}
+		// for kic drivers we use 127.0.0.1 instead of node IP,
+		// because of Docker on MacOs limitations for reaching to container's IP.
+		addr := fmt.Sprintf("https://" + net.JoinHostPort(oci.DefaultBindIPV4, strconv.Itoa(p)))
+		if cc.KubernetesConfig.APIServerName != constants.APIServerName {
+			addr = strings.Replace(addr, oci.DefaultBindIPV4, cc.KubernetesConfig.APIServerName, -1)
```

Comment:
Please do not use string replacement. It makes the code more difficult to follow and adds room for bugs, as we've already seen. What do you think of something like:

```
hostname := h.Driver.GetIP()
port := n.Port

if driver.IsKIC() {
  hostname := oci.DefaultBindIPV4
  port := ...
}

// Warning: this may be an arbitrary string
if  cc.KubernetesConfig.APIServerName != constants.APIServerName {
  hostname =  cc.KubernetesConfig.APIServerName
}

return fmt.Sprintf("https://" + net.JoinHostPort(hostname, strconv.Itoa(port)))
```